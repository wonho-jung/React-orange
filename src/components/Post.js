import React from 'react';
import { markdown } from 'markdown';
function Post({ data, title, body, author, publishData }) {
  console.log(markdown.toHTML(body));
  return (
    <div>
      <h1>Title:{title}</h1>
      <h1>Author:{author}</h1>
      <h1>Date:{publishData}</h1>
      <div
        dangerouslySetInnerHTML={{
          __html: markdown.toHTML(body),
        }}
      />
    </div>
  );
}

export default Post;
