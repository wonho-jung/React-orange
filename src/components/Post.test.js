import React from 'react';
import renderer from 'react-test-renderer';
import Post from './Post';

describe('Post component', () => {
  it('matchs the snapshot', () => {
    const tree = renderer
      .create(
        <Post
          data={'newdata'}
          setNewdata={'setNewdata'}
          key={'id'}
          title={'title'}
          body={'body'}
          author={'author'}
          publishData={'2012'}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
