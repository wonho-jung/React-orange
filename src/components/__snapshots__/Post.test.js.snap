// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component matchs the snapshot 1`] = `
<div
  className="sc-bdnxRM jBSNjv"
>
  <h1
    className="fontWhite cursor"
    onClick={[Function]}
  >
    Title:
    title
  </h1>
  <h3
    className="fontWhite cursor"
    onClick={[Function]}
  >
    Author:
    author
  </h3>
  <p
    className="fontWhite"
  >
    Date:
    2012
  </p>
  <h4
    className="fontWhite"
  >
     Summary: 
    ody
  </h4>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>body</p>",
      }
    }
    style={
      Object {
        "display": "none",
      }
    }
  />
</div>
`;
